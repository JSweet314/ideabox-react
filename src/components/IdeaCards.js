import React from 'react';
import IdeaCard from './IdeaCard.js';
import PropTypes from 'prop-types';
import '../styles/IdeaCards.css';

const IdeaCards = (
  { ideas, updateIdeaQuality, removeIdea, searchValue, qualityFilter}
) => {
  const visibleIdeas = ideas.filter(idea => {
    return (idea.title.includes(searchValue)
      || idea.body.includes(searchValue))
      && idea.quality.includes(qualityFilter);
  }).map(idea => <IdeaCard
    key={idea.id}
    {...idea}
    removeIdea={removeIdea}
    updateIdeaQuality={updateIdeaQuality}
  />
  );

  return (
    <div className="card-container">      
      {visibleIdeas}
    </div>
  );
};

IdeaCards.defaultProps = {
  ideas: []
};

IdeaCards.propTypes = {
  ideas: PropTypes.arrayOf(PropTypes.object),
  searchValue: PropTypes.string.isRequired,
  qualityFilter: PropTypes.string.isRequired,
  removeIdea: PropTypes.func.isRequired,
  updateIdeaQuality: PropTypes.func.isRequired
};

export default IdeaCards;