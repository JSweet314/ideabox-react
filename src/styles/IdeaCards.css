.card-container {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.search {
  font-family: 'Open Sans', sans-serif;
  margin: 2rem 0 0 0;
  height: 4rem;
  padding: 0 0 0 1rem;
  width: 50%;
  border-style: unset;
  border: 2px solid lightgray;
  font-size: 2rem;
}

.search::placeholder {
  font-family: 'Open Sans', sans-serif;
}