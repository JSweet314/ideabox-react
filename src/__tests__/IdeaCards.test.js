import React from 'react';
import {shallow} from 'enzyme';
import IdeaCards from '../components/IdeaCards.js';

describe('IdeaCards', () => {
  let ideaCards, idea1, idea2, visibleIdeas;

  beforeEach(() => {
    idea1 = {
      id: 1234,
      title: 'Hello',
      body: 'World!',
      quality: 'plausible'
    };

    idea2 = {
      id: 4321,
      title: 'New App!',
      body: 'Woo Hoo',
      quality: 'genius'
    };

    visibleIdeas = [idea1, idea2];
    
    ideaCards = shallow(
      <IdeaCards
        visibleIdeas={visibleIdeas}
        updateIdeaQuality={jest.fn()}
        removeIdea={jest.fn()}
        searchIdeas={jest.fn()}
        filterByQuality={jest.fn()}
      />
    );
  });

  it('should render our good friend, IdeaCards', () => {
    expect(ideaCards).toMatchSnapshot();
    
    ideaCards = shallow(
      <IdeaCards
        visibleIdeas={[]}
        updateIdeaQuality={jest.fn()}
        removeIdea={jest.fn()}
        searchIdeas={jest.fn()}
        filterByQuality={jest.fn()}
      />
    );

    expect(ideaCards).toMatchSnapshot();
  });

  it('should have a prop visibleIdeas as well as methods from the app component', () => {
    expect(ideaCards.props('visibleIdeas')).toBeDefined();
    expect(ideaCards.props('updateIdeaQuality')).toBeDefined();
    expect(ideaCards.props('removeIdea')).toBeDefined();
    expect(ideaCards.props('searchIdeas')).toBeDefined();
  });
});
