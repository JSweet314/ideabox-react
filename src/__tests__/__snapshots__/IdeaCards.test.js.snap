// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaCards should render just the Quality Filter and search input if there are no cards to display 1`] = `
<div
  className="card-container"
/>
`;

exports[`IdeaCards should render our good friend, IdeaCards 1`] = `
<div
  className="card-container"
/>
`;
